pyiode/pyiode_exec.pyx

```diff
@@ -6,18 +6,109 @@
+            
+            Parameters
+                Execution sample : 1960Y1:2015Y1
+            
+                Variables loaded
+                    From WS : AOUC AOUC_ COTRES DEBT DTF DTH EFMY EFXY EX FLG FLGR GAP2 GAP_ GOSF HF  
+                    
+                Scalars loaded
+                    From ../data/fun : gamma gamma2 gamma3 gamma4 knf3
+            
+    Examples
+    --------
+        # define the "trace" file
+        iode.w_dest("test_iode.htm", iode.W_HTML)
+    
+        # Simple case: no vars in WS, sample given
+        iode.ws_clear_var()
+        iode.ws_sample_set("1960Y1", "2015Y1")
+        iode.ws_load_idt(f"{IODE_DATA_DIR}fun")
+        iode.idt_execute("1961Y1:2000Y1", "AOUC", f"{IODE_DATA_DIR}fun", trace = 1)
+        AOUC = iode.get_var("AOUC")
+        test_eq('iode.idt_execute("1961Y1:2000Y1", "AOUC", f"{IODE_DATA_DIR}fun, trace = 1") -> AOUC[1961Y1]', 0.24783192, AOUC[1])
+    
+        # Load WS + change value before execution to check the result
+        iode.ws_load_var(f"{IODE_DATA_DIR}fun")
+        AOUC = iode.get_var("AOUC")
+        AOUC[1] = 0.1 
+        iode.set_var("AOUC", AOUC) # var changed in IODE WS
+        iode.idt_execute("1961Y1:2015Y1", "AOUC")
+        AOUC = iode.get_var("AOUC")
+        test_eq('2. iode.idt_execute("1961Y1:2015Y1", "AOUC") -> AOUC[1961Y1]', 0.24783192, AOUC[1])
+    
+        # Shortest way using get_var_as_ndarray(..., False) => no iode.set_var() needed
+        iode.ws_load_var(f"{IODE_DATA_DIR}fun")
+        AOUC = iode.get_var_as_ndarray("AOUC", False)
+        AOUC[1] = 0.1
+        iode.idt_execute("1961Y1:2015Y1", "AOUC")
+        test_eq('3. iode.idt_execute("1961Y1:2015Y1", "AOUC") -> AOUC[1961Y1]', 0.24783192, AOUC[1])
+    
+        # Alternative ways to call idt_execute
+        AOUC = iode.get_var_as_ndarray("AOUC", False)
+        AOUC[1] = 0.1
+        iode.idt_execute(["1961Y1", "2015Y1"], ["AOUC", "FLGR"])
+        test_eq('4. iode.idt_execute(["1961Y1", "2015Y1"], ["AOUC", "FLGR"]) -> AOUC[1961Y1]', 0.24783192, AOUC[1])
+    
+        # Call with empty parameters 
+        iode.ws_load_scl(f"{IODE_DATA_DIR}fun")
+        iode.delete_idt("SSFFX") # SSFFX contains non existent variables
+        AOUC = iode.get_var_as_ndarray("AOUC", False)
+        print(f"AOUC[1]={AOUC[1]}")
+        AOUC[1] = 0.1
+        iode.idt_execute() # all idts on full sample using current loaded WS
+        AOUC = iode.get_var_as_ndarray("AOUC", False)
+        test_eq(' 5. iode.idt_execute() -> AOUC[1961Y1]', 0.24677525, AOUC[1])
+    
+        iode.w_close()
+    
     '''
-    pass
+
     
+    # Transforms Lists into strs
+    sample = arg_to_str(sample)
+    idt_list = arg_to_str(idt_list, ";")
+    var_files = arg_to_str(var_files, ";")
+    scl_files = arg_to_str(scl_files, ";")
+
+    clear_error_msgs() # clear messages before executing the function
+    rc = IodeExecuteIdts(cstr(sample), cstr(idt_list), cstr(var_files), cstr(scl_files), trace) 
```

Comment:
A C++ function already exists for that. That is what I was talking about when I said there are lot og duplicate code between the C++ API and b_api.c :\

But let use `IodeExecuteIdts` for the moment. I will replace it by a call to the C++ function later.